src/python/pants/init/extension_api.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from collections.abc import Callable, Iterable
+from typing import Any, Protocol
+
+from pants.build_graph.build_file_aliases import BuildFileAliases
+from pants.engine.rules import Rule
+from pants.engine.target import Target
+from pants.engine.unions import UnionRule
+from pants.goal.auxiliary_goal import AuxiliaryGoal
+from pants.option.subsystem import Subsystem
+
+
+class ExtensionInitContextV0(Protocol):
+    """A context object used by extension initializers to register their capabilities with the
+    engine."""
```

Comment:
I still need to add an API for ensuring plugin load order. I'm thinking "check_plugin_loaded_after` (or similar name).

Reply:
`dependencies` ? :)